src/test/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionListTest.java

```diff
@@ -79,4 +99,113 @@
+            */
+            waitFor(logging::getMessages, hasItem(containsString("Will resume [org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep")));
+            WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+            SemaphoreStep.success("wait/1", null);
+            WorkflowRun b = p.getBuildByNumber(1);
+            r.waitForCompletion(b);
+            r.assertBuildStatus(Result.SUCCESS, b);
+        });
+    }
+
+    @Issue("TODO")
+    @Test public void resumeStepExecutions() throws Throwable {
+        sessions.then(r -> {
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("noResume()", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            r.waitForMessage("Starting non-resumable step", b);
+            // TODO: Unclear how this might happen in practice.
+            FlowExecutionList.get().unregister(b.asFlowExecutionOwner());
+        });
+        sessions.then(r -> {
+            WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+            WorkflowRun b = p.getBuildByNumber(1);
+            r.waitForCompletion(b);
+            r.assertBuildStatus(Result.FAILURE, b);
+            r.assertLogContains("Unable to resume NonResumableStep", b);
+        });
+    }
+
+    public static class NonResumableStep extends Step implements Serializable {
+        public static final long serialVersionUID = 1L;
+        @DataBoundConstructor
+        public NonResumableStep() { }
+        @Override
+        public StepExecution start(StepContext sc) throws Exception {
+            return new ExecutionImpl(sc);
+        }
+
+        private static class ExecutionImpl extends StepExecution implements Serializable {
+            public static final long serialVersionUID = 1L;
+            private ExecutionImpl(StepContext sc) {
+                super(sc);
+            }
+            @Override
+            public boolean start() throws Exception {
+                getContext().get(TaskListener.class).getLogger().println("Starting non-resumable step");
+                return false;
+            }
+            @Override
+            public void onResume() {
+                getContext().onFailure(new AbortException("Unable to resume NonResumableStep"));
+            }
+        }
+
+        @TestExtension public static class DescriptorImpl extends StepDescriptor {
+            @Override
+            public Set<? extends Class<?>> getRequiredContext() {
+                return Collections.singleton(TaskListener.class);
+            }
+            @Override
+            public String getFunctionName() {
+                return "noResume";
+            }
+        }
+    }
+
+    /**
+     * Wait up to 5 seconds for the given supplier to return a matching value.
+     */
+    private static <T> void waitFor(Supplier<T> valueSupplier, Matcher<T> matcher) throws InterruptedException {
```

Comment:
or add a test dep on Awaitility

Reply:
For something small and one-off like this I prefer to avoid adding a new dependency.